app/script/conversation/ConversationRepository.coffee

```diff
@@ -219,12 +219,16 @@ class z.conversation.ConversationRepository
   @param message_id [String]
   @return [Promise] z.entity.Message
   ###
-  get_message_from_db: (conversation_et, message_id) =>
-    @conversation_service.load_event_from_db conversation_et.id, message_id
-    .then (event) =>
-      if event
-        return @event_mapper.map_json_event event, conversation_et
-      throw new z.conversation.ConversationError z.conversation.ConversationError::TYPE.MESSAGE_NOT_FOUND
+  get_message_in_conversation_by_id: (conversation_et, message_id) =>
+    message_et = conversation_et.get_message_by_id message_id
+    if message_et
+      Promise.resolve message_et
+    else
```

Comment:
you could also just return before the else to avoid the indentation as it's fairly straight forward